packages/flutter/lib/src/rendering/slivers.dart

```diff
@@ -0,0 +1,1850 @@
+        if (owner != null && owner.debugDoingLayout)
+          hint = 'Only the object itself can set its geometry. It is a contract violation for other objects to set it.';
+      }
+      if (sizedByParent)
+        contract = 'Because this RenderSliver has sizedByParent set to true, it must set its geometry in performResize().';
+      else
+        contract = 'Because this RenderSliver has sizedByParent set to false, it must set its geometry in performLayout().';
+      throw new FlutterError(
+        'RenderSliver geometry setter called incorrectly.\n'
+        '$violation\n'
+        '$hint\n'
+        '$contract\n'
+        'The RenderSliver in question is:\n'
+        '  $this'
+      );
+    });
+    _geometry = value;
+  }
+
+  @override
+  Rect get semanticBounds => paintBounds;
+
+  @override
+  Rect get paintBounds {
+    assert(constraints.axis != null);
+    switch (constraints.axis) {
+      case Axis.horizontal:
+        return new Rect.fromLTWH(
+          0.0, 0.0,
+          constraints.crossAxisExtent,
+          geometry.paintExtent
+        );
+      case Axis.vertical:
+        return new Rect.fromLTWH(
+          0.0, 0.0,
+          geometry.paintExtent,
+          constraints.crossAxisExtent
+        );
+    }
+    return null;
+  }
+
+  @override
+  void debugResetSize() { }
+
+  @override
+  void debugAssertDoesMeetConstraints() {
+    assert(geometry.debugAssertIsValid);
+    assert(() {
+      if (geometry.paintExtent > constraints.remainingPaintExtent) {
+        throw new FlutterError(
+          'SliverGeometry has a paintOffset that exceeds the remainingPaintExtent from the constraints.\n'
+          'The render object whose geometry violates the constraints is the following:\n'
+          '  ${toStringShallow('\n  ')}\n' +
+          _debugCompareFloats('remainingPaintExtent', constraints.remainingPaintExtent,
+                              'paintExtent', geometry.paintExtent) +
+          'The paintExtent must cause the child sliver to paint within the viewport, and so '
+          'cannot exceed the remainingPaintExtent.'
+        );
+      }
+      return true;
+    });
+  }
+
+  @override
+  void performResize() {
+    assert(false);
+  }
+
+  /// For a center sliver, the distance before the absolute zero scroll offset
+  /// that this sliver can cover.
+  ///
+  /// For example, if an [AxisDirection.down] viewport with an
+  /// [RenderViewport2.anchor] of 0.5 has a single sliver with a height of 100.0
+  /// and its [centerOffsetAdjustment] returns 50.0, then the sliver will be
+  /// centered in the viewport when the scroll offset is 0.0.
+  ///
+  /// The distance here is in the opposite direction of the
+  /// [RenderViewport2.axisDirection], so values will typically be positive.
+  double get centerOffsetAdjustment => 0.0;
+
+  @override
+  void performLayout();
```

Comment:
Removed. I had copied in the interface and just forgot to remove the ones I didn't want to reimplement yet.